infinitory/errors.py

```diff
@@ -0,0 +1,137 @@
+import os
+import pickle
+import sys
+import time
+
+from datetime import datetime
+
+import logging
+
+
+class ErrorParser(object):
+    def __init__(self):
+        self._reports = dict()
+        self._all_errors = []
+        self._unique_errors = []
+
+        self._logger = logging.getLogger()
+
+    def delete_report_cache(self):
+        cache_path = './cache/'
+        for f in os.listdir(cache_path):
+            # Delete cache item if older than 2 hours
+            if os.stat(os.path.join(cache_path,f)).st_mtime < time.mktime(datetime.now().timetuple()) - (1 * 3600):
+                print("Deleting File " + os.path.join(cache_path,f))
+                os.remove(os.path.join(cache_path,f))
+
+    def load_reports_fast(self, pupdb):
+        # Takes Much Longer
+        self.reports = dict()
+        for report in pupdb.query('reports[] { hash in nodes[latest_report_hash] { } }'):
+            if report['status'] == 'failed':
+                self.reports[report['certname']] = report
+
+    def load_reports(self, pupdb):
+        for report in pupdb.query('nodes[certname, latest_report_hash] { }'):
+            cache_file = "./cache/%s" % report["latest_report_hash"]
+            if os.path.isfile(cache_file):
+                full_report = pickle.load(open(cache_file, "rb"))
+                sys.stdout.write('#')
+            else:
+                full_report = pupdb.query('reports[] { hash = "' + report["latest_report_hash"] + '" }')
+                pickle.dump( full_report, open(cache_file, "wb" ) )
+                sys.stdout.write('.')
+            sys.stdout.flush()
+
+            self._reports[report["certname"]] = full_report[0]
+
+    def stored_errors(self):
+        return [
+            "Could not retrieve catalog from remote server: Error 500 on SERVER: Server Error: Evaluation Error: Error while evaluating a Function Call, Untrusted facts (left) don't match values from certname (right)"
+        ]
+
+    def matches_stored_error(self, message):
+        for se in self.stored_errors():
+            if message.startswith(se):
+                return True, se
```

Comment:
Make this return the error string, or `None` if it's not found. It's more idiomatic Python.

Reply:
Will do